signature/src/lib.rs

```diff
@@ -30,3 +35,377 @@ pub extern "C" fn verify_rsa(
         Ok(_) => true,
     }
 }
+
+#[no_mangle]
+pub extern "C" fn verify_balrog(
```

Comment:
Consider adding header comments for the new code. Also, is something like verify_signature a better name?

Reply:
It very well might be, technically this is a validation of Mozilla's content signature algorithm, it just so happens that "Balrog" is the name of the service that uses the content signature.